cocos/rendering/custom/binary-archive.ts

```diff
@@ -64,10 +64,11 @@ export class BinaryInputArchive implements InputArchive {
     }
     readString (): string {
         const length = this.readNumber();
-        const value = new Uint8Array(this.dataView.buffer, this.offset, length).toString();
+        const value = new Uint8Array(this.dataView.buffer, this.offset, length);
         this.offset += length;
-        return value;
+        return this.textDecoder.decode(value);
     }
     offset = 0;
     dataView: DataView;
+    textDecoder = new TextDecoder('utf-8');
```

Comment:
Do all platforms have this object?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder
It seems working on platforms.